packages/features/bookings/lib/getBookingFields.ts

```diff
@@ -0,0 +1,333 @@
+  const parsedBookingFields = eventTypeBookingFields.parse(bookingFields || []);
+  const parsedCustomInputs = customInputSchema.array().parse(customInputs || []);
+  workflows = workflows || [];
+  return ensureBookingInputsHaveSystemFields({
+    bookingFields: parsedBookingFields,
+    disableGuests,
+    additionalNotesRequired: parsedMetaData?.additionalNotesRequired || false,
+    customInputs: parsedCustomInputs,
+    workflows,
+  });
+};
+
+export const ensureBookingInputsHaveSystemFields = ({
+  bookingFields,
+  disableGuests,
+  additionalNotesRequired,
+  customInputs,
+  workflows,
+}: {
+  bookingFields: Fields;
+  disableGuests: boolean;
+  additionalNotesRequired: boolean;
+  customInputs: z.infer<typeof customInputSchema>[];
+  workflows: Prisma.EventTypeGetPayload<{
+    select: {
+      workflows: {
+        select: {
+          workflow: {
+            select: {
+              id: true;
+              steps: true;
+            };
+          };
+        };
+      };
+    };
+  }>["workflows"];
+}) => {
+  // If bookingFields is set already, the migration is done.
+  const handleMigration = !bookingFields.length;
+  const CustomInputTypeToFieldType = {
+    [EventTypeCustomInputType.TEXT]: BookingFieldType.text,
+    [EventTypeCustomInputType.TEXTLONG]: BookingFieldType.textarea,
+    [EventTypeCustomInputType.NUMBER]: BookingFieldType.number,
+    [EventTypeCustomInputType.BOOL]: BookingFieldType.boolean,
+    [EventTypeCustomInputType.RADIO]: BookingFieldType.radio,
+    [EventTypeCustomInputType.PHONE]: BookingFieldType.phone,
+  };
+
+  const smsNumberSources = [] as NonNullable<typeof bookingFields[number]["sources"]>;
+  workflows.forEach((workflow) => {
+    workflow.workflow.steps.forEach((step) => {
+      if (step.action === "SMS_ATTENDEE") {
+        const workflowId = workflow.workflow.id;
+        smsNumberSources.push(
+          getSmsReminderNumberSource({
+            workflowId,
+            isSmsReminderNumberRequired: !!step.numberRequired,
+          })
+        );
+      }
+    });
+  });
+
+  // These fields should be added before other user fields
+  const systemBeforeFields: typeof bookingFields = [
+    {
+      defaultLabel: "your_name",
+      defaultPlaceholder: "example_name",
+      type: "name",
+      name: "name",
+      required: true,
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+    {
+      defaultLabel: "email_address",
+      defaultPlaceholder: "you@example.com",
+      type: "email",
+      name: "email",
+      required: true,
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
+    {
+      defaultLabel: "location",
+      type: "radioInput",
+      name: "location",
+      // Even though it should be required it is optional in production with backend choosing CalVideo as the fallback
```

Comment:
@CarinaWolli Yes, that should happen already. It is handled here https://github.com/calcom/cal.com/pull/6560/files#diff-d360af68e64bd72af3fbe53bea6bdfbf7a70e0e1366670d6adfd972c1059f88bR151

Reply:
I meant in the event-type settings. So this one: 
![Screenshot 2023-02-15 at 10 46 39](https://user-images.githubusercontent.com/30310907/219078154-4812488b-2f8a-4797-a5a0-4c11dbd3abcd.png)

But it might be ok to still have it here, was just a thought

Reply:
Oh, so it should remain I think. The plan is to move location select to the right sidebar where we show locations anyway. There we always show the location if there is just one